R/drive_download.R

```diff
@@ -0,0 +1,72 @@
+#' Download a file from Google Drive.
+#'
+#' @template file
+#' @param out_path Character. Path for output file. Google files can
+#'   be downloaded to types specified [here](https://developers.google.com/drive/v3/web/manage-downloads).
+#' @param overwrite A logical scalar, do you want to overwrite a file that already
+#'    exists locally?
+#' @template verbose
+#'
+#' @examples
+#' \dontrun{
+#' ## Save "chickwts.csv" to the working directory as "chickwts.csv".
+#' drive_download(file = "chickwts.csv", out_path = "chickwts.csv")
+#'
+#' ## Save Google Sheet named "chickwts" to the working directory as
+#' ## "chickwts.csv".
+#' drive_download(file = "chickwts", out_path = "chickwts.csv")
+#'}
+#' @export
+drive_download <- function(file = NULL,
+                           out_path = NULL,
+                           overwrite = FALSE,
+                           verbose = TRUE) {
+  in_file <- as_dribble(file)
+
+  if (nrow(in_file) == 0L) {
```

Comment:
Perhaps that means we should improve the messaging in those checkers. Or allow a message to be provided in the call, so that you can override with something more informative, when possible and important? You could open an issue for that, but go ahead and simplify things here.